subsys/nrf_security/src/drivers/cracen/cracenpsa/src/key_management.c

```diff
@@ -672,7 +674,11 @@ static psa_status_t export_ecc_public_key_from_keypair(const psa_key_attributes_
 			}
 			break;
 		case PSA_ECC_FAMILY_TWISTED_EDWARDS:
-			if (key_bits_attr == 255) {
+			if (key_alg == PSA_ALG_ED25519PH) {
+				priv_key.def = si_sig_def_ed25519ph;
+				priv_key.key.ed25519 = (struct sx_ed25519_v *)key_buffer;
+				pub_key.key.ed25519 = (struct sx_ed25519_pt *)data;
+			} else if (key_alg == PSA_ALG_PURE_EDDSA) {
```

Comment:
Yes, both PSA_ALG_ED25519PH and PSA_ALG_PURE_EDDSA have key_bits_attr=255. So if left it would always default to whoever was first

Reply:
I mean, the condition was not for anything else than `PSA_ALG_ED25519PH` and `PSA_ALG_PURE_EDDSA`? (Just checking that `key_bits_attr == 255` wasn't true for another case which isn't one of those two algs, in which case it would now be false).

Reply:
The condition is for which edwards curve based algorithm you use. Previously it separated on key_bits because ed25519 has 255 and ed448 has 456. Since I added in PH it needed to be changed since you can't separate those two based on that. I removed the check since there is no need to check for key_bits if you check which alg it is(as far as I know generally, but in this specific case certainly)

Reply:
Alright, makes sense!

Reply:
The thing is that ED448 will also use the PSA_ALG_PURE_EDDSA.
https://arm-software.github.io/psa-api/crypto/1.2/api/ops/sign.html#c.PSA_ALG_PURE_EDDSA

So I think that this will break 448 support actually.  You should probably check the key bits inside the PURE_EDDSA case here.

Reply:
Aah that makes more sense then. It would break 448 support. But that does make me wonder if ed488 is already slightly broken as in sign.c:122 it checks for pure_eddsa and then set the privkey to si_sig_def_ed25519. So signs on it would be ed25519 signs and not 448?

Reply:
It seems that you are right and it had issues before indeed.